modules/cnf-topology-aware-lifecycle-manager-precache-user-spec-images.adoc

```diff
@@ -0,0 +1,165 @@
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `platformImage`, `operatorsIndexes`, and the `operatorsPackagesAndChannels` fields are automatically populated from the policies of the managed clusters.
+<2> Specifies the minimum required disk space on the cluster in Gigabytes. If unspecified, {cgu-operator} defines a default value for {product-title} images.
+<3> Specifies the list of additional images to pre-cache.
+
+// Procedure
+
+[IMPORTANT]
+====
+Once you install the images on the cluster, you cannot modify or delete them.
+====
+
+. Create a `PreCachingConfig` CR with the list of additional images you want to pre-cache:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: PreCachingConfig
+metadata:
+  name: exampleconfig
+  namespace: default <1>
+spec:
+  overrides:
+    platformImage: quay.io/openshift-release-dev/ocp-release@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30Gi
+  excludePrecachePatterns:
+    - aws
+    - vsphere
+  additionalImages:
+    - quay.io/exampleconfig/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `namespace` must be accessible to the hub cluster.
+
+. Create a `ClusterGroupUpgrade` CR with the `preCaching` field set to `true` and specify the `PreCachingConfig` CR created in the previous step:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+  namespace: default
+spec:
+  clusters:
+  - sno1
+  - sno2
+  preCaching: true
+  preCachingConfigRef:
+  - name: exampleconfig
+    namespace: default
+  remediationStrategy:
+    timeout: 240
+----
+
+. {cgu-operator} verifies the `ClusterGroupUpgrade` CR.
+If the verification is successful, the pre-caching reconciliation starts.
++
+[NOTE]
+====
+All sites are pre-cached concurrently.
+====
+// What happens if the verification fails? Any common mistakes? Troubleshooting steps?
+
+.Verification
+// This is all automatic. But what if something goes wrong? Is this all automatically printed or do you have to request?
+
+. The pre-caching configurations are validated by checking if the managed policies exist.
+A valid configuration results in the following status:
```

Comment:
1) What happens if the verification fails? Any common mistakes? Troubleshooting steps that we should share?
2) Is this all automatic or do we have to print it out with `oc get cgu`?

Reply:
Pre-caching configs are validated against the PreCachingConfig CRD upon creation. After successful creation, the referenced PreCachingConfig CR in the CGU is validated as previously with the additional validation step of checking the new `SpaceRequired` parameter. It is important to note that the PreCachingConfig CR does not reference any policies - this is accomplished by the CGU CR. 

The usual procedure of `oc get cgu` is sufficient to view the status of the pre-caching job as well as to troubleshoot. If an error occurs (as will be shown in the `.status.conditions` and `.status.precaching.status` fields), the logs of the TALM pod (specifically the manager container) can be inspected to get more information as to the cause of failure.